envoy: Bump envoy to v1.25.4

sayboras opened the pull request:
This is the latest envoy version, just for testing purpose.

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/master/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

sayboras committed:
envoy: Bump envoy to v1.25.4

The image is built from below GHA

https://github.com/cilium/proxy/actions/runs/4625251879/jobs/8181060088

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
vendor: Update cilium/proxy version to latest

Signed-off-by: Tam Mach <tam.mach@cilium.io>

jrajahalme commented on pkg/envoy/accesslog_server.go:
```diff
@@ -164,11 +164,11 @@ func logRecord(pblog *cilium.LogEntry) *logger.LogRecord {
 	} else {
 		// Default to the deprecated HTTP log format
```
IMO we should just drop this block altogether. If `pblog.GetHttp()` is non-nil we already take the first `if` statement above.

jrajahalme reviewed (changes requested):
Better remove the stale accesslog server code block.

sayboras committed:
build: Combine sed statements for ENVOY sha

Relates: https://github.com/cilium/cilium/pull/24750#pullrequestreview-1372739218
Signed-off-by: Tam Mach <tam.mach@cilium.io>

meyskens reviewed (approved):
LGTM
